pkg/util/ip_reservation_test.go

```diff
@@ -0,0 +1,335 @@
+		i++
+		ipRangeString := fmt.Sprint(cidrIP.String(), "/", ipRangeSize)
+		ipRanges[ipRangeString] = true
+	}
+	if err != nil {
+		t.Errorf(err.Error())
+	} else if i != ipRangesCount {
+		t.Errorf("The required number of IP ranges %d are not available in the CIDR %s", ipRangesCount, cidr)
+	}
+	return ipRanges
+}
+
+func TestValidateCIDROverlap(t *testing.T) {
+	cases := []struct {
+		name          string
+		cidr1         string
+		cidr2         string
+		expected      bool
+		errorExpected bool
+	}{
+		{
+			name:          "Overlapping CIDRs",
+			cidr1:         "192.168.92.0/29",
+			cidr2:         "192.168.92.48/26",
+			expected:      true,
+			errorExpected: false,
+		},
+		{
+			name:          "Non overlapping CIDRs",
+			cidr1:         "192.168.92.0/29",
+			cidr2:         "192.168.22.67/26",
+			expected:      false,
+			errorExpected: false,
+		},
+		{
+			name:          "Non overlapping CIDRs with same cidr size",
+			cidr1:         "192.168.92.247/29",
+			cidr2:         "192.168.92.248/29",
+			expected:      false,
+			errorExpected: false,
+		},
+		{
+			name:          "Overlapping CIDRs with same cidr size",
+			cidr1:         "192.168.92.249/29",
+			cidr2:         "192.168.92.255/29",
+			expected:      true,
+			errorExpected: false,
+		},
+		{
+			name:          "Invalid CIDR provided",
+			cidr1:         "192.168.92.0",
+			cidr2:         "192.168.22.67/26",
+			errorExpected: true,
+		},
+	}
+
+	for _, test := range cases {
+		_, ipnet1, _ := net.ParseCIDR(test.cidr1)
+		_, ipnet2, _ := net.ParseCIDR(test.cidr2)
+		overlap, err := isOverlap(ipnet1, ipnet2)
+		if err != nil && !test.errorExpected {
+			t.Errorf("test %q failed: got error %s, expected cidr overlap between %s and %s to be %t", test.name, err.Error(), test.cidr1, test.cidr2, test.expected)
+		} else if err == nil && test.errorExpected {
+			t.Errorf("test %q failed: got cidr overlap value %t, expected error", test.name, overlap)
+		} else if !test.errorExpected && overlap != test.expected {
+			t.Errorf("test %q failed: got overlap for cidr %s and %s as %t, expected %t", test.name, test.cidr1, test.cidr2, test.expected, test.expected)
+		}
+	}
+}
+
+func TestIncrementIP(t *testing.T) {
+
+	cases := []struct {
+		name          string
+		currentIP     string
+		step          byte
+		expected      string
+		errorExpected bool
+	}{
+		{
+			name:          "Valid IP increment without carry forward to significant bytes",
+			currentIP:     "192.168.92.32",
+			step:          143,
+			expected:      "192.168.92.175",
+			errorExpected: false,
+		},
+		{
+			name:          "Valid increment with carry forward to significant bytes with maximum step size",
+			currentIP:     "192.255.255.253",
+			step:          255,
+			expected:      "193.0.0.252",
+			errorExpected: false,
+		},
+		{
+			name:          "Valid increment with carry forward to significant bytes",
```

Comment:
Can you add more test cases with step 8 to potentially cover off-by-one errors, and also carry forward at each of the bytes?